File: BloodBankMSApi/Models/BloodDonorDonation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BloodBankMSApi.Models
{
    public class BloodDonorDonation
    {

        public int Id { get; set; }
        
        [Required]
        public int BloodDonorId { get; set; }
        // [ForeignKey]

        public virtual BloodDonor? BloodDonor {get;set;}

        [Required]
        [DataType(DataType.Date)]
        public DateTime BloodDonationDate { get; set; }
        
        [Required]
        [Range (0, int.MaxValue)]
        public int NumberofBottle { get; set; }
        
        [Required]
        [Range(0, int.MaxValue)]
        public double Weight { get; set; }
        
        [Required]
        [Range(0, int.MaxValue)]
        public double HBCount { get; set; }

        [Required]
        public int BloodDonationCampId { get; set; }
        public virtual BloodDonationCamp? BloodDonationCamp { get; set; }
        [Required]
        public DateTime ExpiryDate { get; set; }

        [ForeignKey("BloodBank")]
        public int BloodBankId { get; set; }
    }
}

File: BloodBankMS_App/BloodBankApp/src/app/Models/blood-donor-donation.ts
// export class BloodDonorDonation {
//     id!:number;
//     bloodDonorId!:number;
//     bloodDonationDate!:Date;
//     numberofBottle!:number;
//     weight!:number;
//     hBCount!:number;
//     bloodDonationCampId!:number;
//     expiryDate!:Date;
//     bloodBankId!:number;
// }
import { BloodBank } from "./blood-bank";
import { BloodDonationCamp } from "./blood-donation-camp";
import { BloodDonor } from "./blood-donor";
 
export class BloodDonorDonation {
    id!:number;
    bloodDonorId!:number;
    bloodDonor!:BloodDonor;
    bloodDonationDate!:Date;
    numberofBottle!:number;
    weight!:number;
    hbCount!:number;
    bloodDonationCampId!:number;
    bloodDonationCamp!:BloodDonationCamp;
    expiryDate!:Date;
    bloodBankId!:number;
    bloodBank!:BloodBank;
}